### ISSUE TITLE
[LOCKED] Why it is impossible to support every OS updates

### ISSUE THREAD

**Issue by kyujin-cho** [kyujin-cho/pixel-volte-patch#303]
### TL;DR
Of course, this application is not a magic wand for your cellular connection!

## Technical details about enabling VoLTE on Android
Integrating VoLTE (or VoWiFi) support onto the Android system generally consists of two big chunks: adding IMS profile to the modem firmware, and then appending the carrier information as supported operator inside Android's own IMS whitelist. Both jobs require attention from both carrier and device vendor, but not Google itself (in terms of OS development firm). 
In theory, actual implementations of the whole IMS process in the modem firmware part may vary for every mobile network operators. But in general most of the carriers tend to fork some of the famous implementations. On the normal condition, when the OS boots and tries to open the cellular connection, the system first reads array of items declared as "VoLTE supported carriers" from the vendor-specific application and compares with the cellular operator the SIM tries to connect with (exact idea goes through when enabling VoWiFi feature). If they find a match then the system will try to always utilize IMS whenever you try to call or text somebody - otherwise Android will just forgive on establishing IMS connections and fallback to CSFB operations. 

## What this application covers
Pixel IMS tries to dig in the latter part by force enabling the flag to determine OS' effort to use IMS, regardless of the whitelist written by the smartphone vendor. Now here is the magic part: by the fact that most operators share similar IMS implementations, sometimes it just works by even without patching the modem firmware itself. 

### About the "not working since update XX" reports
But the happily-ever-after scenario can always break as soon as modem firmware updates - with some confidential changes there might be chances of the compatibility being broken. This usually is the case where your phone just start calling your friend or colleague with the janky old CSFB method when the normal OS update ships to your phone. Whenever such situation happens then I do not have any more room to solve the problem.

### Then can't you just patch the firmware itself?
That way is a total different idea. Of course it may be possible, but only after managing to reverse engineer the whole firmware file itself. There are rare cases I found of actually accomplishing the theory at some of the Korean forums, but even such solutions still require unlocking the bootloader since it has to reflash whole stock firmware.

## Conclusion
This patch is always an interim solution for your phone, and always the best idea to make those changes work permanently is to pursue your local carrier to adapt IMS on your devices.
